execute/table.go

```diff
@@ -1194,6 +1216,63 @@ func (b *ColListTableBuilder) AppendValue(j int, v values.Value) error {
 	}
 }
 
+func (b *ColListTableBuilder) SetNil(i, j int) error {
+	if j < 0 || j > len(b.cols) {
+		return fmt.Errorf("set nil: column does not exist, index out of bounds: %d", j)
+	}
+	if i < 0 || i > b.cols[j].Len() {
+		return fmt.Errorf("set nil: row  does not exist, index out of bounds: %d", i)
+	}
+
+	if len(b.nils) != len(b.cols) {
```

Comment:
This should be done in `AddCol`. Basically when you add the new column, do `append(b.nils, nil)` to give it a nil map.